django/core/paginator.py

```diff
@@ -99,6 +106,17 @@ def _get_page_range(self):
         return six.moves.range(1, self.num_pages + 1)
     page_range = property(_get_page_range)
 
+    def _is_ordered_queryset(self):
+        """
+        Check if the object list is Queryset and if it's not ordered
+        and wran user
+        """
+        if hasattr(self.object_list, 'ordered')\
+           and not self.object_list.ordered:
+            warnings.warn("Pagination may yield consistent results with"
```

Comment:
sorry I had some typos in my suggestion. It should be "Pagination may yield inconsistent results if object_list isn't ordered". May help to include the repr of the object_list too, e.g. to make the test failures easier to debug.
also, use hanging indent:
```
warnings.warn(
    "..."
)
```